src/queue/eventhub/utils.ts

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable sort-keys */
+import {
+  connect,
+  disconnectWithoutError
+} from "@pagopa/fp-ts-kafkajs/dist/lib/KafkaOperation";
+import {
+  AzureEventhubSas,
+  AzureEventhubSasFromString
+} from "@pagopa/fp-ts-kafkajs/dist/lib/KafkaProducerCompact";
+import { defaultLog } from "@pagopa/winston-ts";
+import * as E from "fp-ts/Either";
+import * as IO from "fp-ts/IO";
+import * as TE from "fp-ts/TaskEither";
+import { constVoid, pipe } from "fp-ts/lib/function";
+import {
+  Consumer,
+  ConsumerConfig,
+  ConsumerRunConfig,
+  ConsumerSubscribeTopics,
+  EachBatchHandler,
+  EachMessageHandler,
+  Kafka,
+  KafkaConfig
+} from "kafkajs";
+
+export interface IKafkaConsumerCompact {
+  readonly consumer: Consumer;
+}
+
+export type KafkaConsumerCompact = IO.IO<IKafkaConsumerCompact>;
+
+export type ValidableKafkaConsumerConfig = KafkaConfig & ConsumerConfig;
+
+export const fromConfig = (
+  config: ValidableKafkaConsumerConfig
+): KafkaConsumerCompact => (): IKafkaConsumerCompact => ({
+  consumer: new Kafka(config).consumer(config)
+});
+
+export const fromSas = (
+  sas: AzureEventhubSas,
+  groupId?: string
+): KafkaConsumerCompact =>
+  pipe(
+    {
+      brokers: [`${sas.url}:9093`],
+      ssl: true,
+      sasl: {
+        mechanism: "plain" as const,
+        username: "$ConnectionString",
+        password: AzureEventhubSasFromString.encode(sas)
+      },
+      clientId: sas.policy,
+      groupId: groupId || "consumer-group",
+      topic: sas.name
+    },
+    fullConfig => fromConfig(fullConfig)
+  );
+
+export const getEventHubConsumer = (
+  connectionString: string
+): E.Either<Error, KafkaConsumerCompact> =>
+  pipe(
+    AzureEventhubSasFromString.decode(connectionString),
+    E.map(sas => fromSas(sas)),
+    E.mapLeft(errors =>
+      pipe(
+        defaultLog.either.error(
+          `Error during decoding EventHub ConnectionURI - ${errors}`
+        ),
+        () => new Error(`Error during decoding Event Hub SAS`)
+      )
+    )
+  );
```

Comment:
```suggestion
    E.bimap(
      errors =>
        pipe(
          defaultLog.either.error(
            `Error during decoding EventHub ConnectionURI - ${errors}`
          ),
          () => new Error(`Error during decoding Event Hub SAS`)
        ),
      fromSas
    )
```
In this case I think it is possible to use E.bimap